subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultMavenArtifactRepository.java

```diff
@@ -239,6 +241,20 @@ public void metadataSources(Action<? super MetadataSources> configureAction) {
         configureAction.execute(metadataSources);
     }
 
+    @Override
+    public void setKind(RepositoryKind kind) {
+        Action<? super ArtifactResolutionDetails> filter = null;
+        switch (kind) {
+            case RELEASES_ONLY:
+                filter = ReleasesOnly.INSTANCE;
+                break;
+            case SNAPSHOTS_ONLY:
```

Comment:
Where is the case for `MIXED` or `BOTH`?

Reply:
We don't need it since `MIXED` means "any mix of modules", so no filter, so `null`.